src/ControlSystem/ExpirationTimes.hpp

```diff
@@ -111,31 +128,69 @@ std::unordered_map<std::string, double> initial_expiration_times(
-      [&initial_time, &measurements_per_update, &domain_creator,
-       &initial_expiration_times](const auto& option_holder) {
-        const auto& controller = option_holder.controller;
-        const std::string& name =
+  using control_systems = tmpl::list<typename OptionHolders::control_system...>;
+
+  // First string is name of each control system. Second string is combination
+  // of control systems with same measurement
+  std::unordered_map<std::string, std::string> map_of_names =
+      system_to_combined_names<control_systems>();
+
+  std::unordered_map<std::string, double> combined_expiration_times{};
+  std::unordered_set<std::string> infinite_expiration_times{};
+  for (const auto& [control_system_name, combined_name] : map_of_names) {
+    (void)control_system_name;
+    if (combined_expiration_times.count(combined_name) != 1) {
+      combined_expiration_times[combined_name] =
+          std::numeric_limits<double>::infinity();
+    }
+  }
+
+  [[maybe_unused]] const auto combine_expiration_times =
+      [&initial_time, &measurements_per_update, &domain_creator, &map_of_names,
+       &combined_expiration_times,
+       &infinite_expiration_times](const auto& option_holder) {
+        const std::string& control_system_name =
             std::decay_t<decltype(option_holder)>::control_system::name();
+        const std::string& combined_name = map_of_names[control_system_name];
+
         auto tuner = option_holder.tuner;
         Tags::detail::initialize_tuner(make_not_null(&tuner), domain_creator,
-                                       initial_time, name);
+                                       initial_time, control_system_name);
 
+        const auto& controller = option_holder.controller;
         const DataVector measurement_timescales =
             calculate_measurement_timescales(controller, tuner,
                                              measurements_per_update);
+        const double min_measurement_timescale = min(measurement_timescales);
+
+        double initial_expiration_time = function_of_time_expiration_time(
+            initial_time, DataVector{1, 0.0},
+            DataVector{1, min_measurement_timescale}, measurements_per_update);
+        initial_expiration_time = option_holder.is_active
+                                      ? initial_expiration_time
+                                      : std::numeric_limits<double>::infinity();
+
+        if (initial_expiration_time ==
+            std::numeric_limits<double>::infinity()) {
+          infinite_expiration_times.insert(control_system_name);
+        }
 
-        const double initial_expiration_time = function_of_time_expiration_time(
-            initial_time, DataVector{measurement_timescales.size(), 0.0},
-            measurement_timescales, measurements_per_update);
         // Don't have to worry about if functions of time are being overridden
         // because that will be taken care of elsewhere.
```

Comment:
Should this comment be removed since #4487?

Reply:
Yup, removed it.